File: crm-portal-ts/src/config_validation.ts
import { config } from "app"; // Assuming app module has a configuration object with ENV and APP_DISPLAY_NAME properties. 
// Also assuming that the DATABASE_URL is of type string, IS_PROD would be boolean (true/false), SECRET_KEY & CRM_* are strings etc...  
import { logger } from "app"; // Assuming app module has a logging object with info method. 
// Also assuming that the _validate* functions use log as their argument and return void or Promise<void> depending on what they do (logging, not doing anything else).   

function reportStartupConfig(): void {  
     logger("app", "STARTUP CONFIGURATION REPORT");  // Logger function in app module. Assuming it logs with a prefix of 'app'.
     
     setLogInfo(`Environment: ${config.ENV}`);   
      
     const dbType = isSQLiteInConfig() ? "SQLite" : "Other";  
     logger("startup_config", `Database type:  ${dbType}`)  // Logger function in startup config module (assuming it logs with a prefix of 'startup').   
     
       _validateSecurity();         
       
       if(isTerytWSUserConfigured()){  
           logger("teryt", `WebService user configured: ${config.TERYT_WS_USER}`)  // Logger function in teryt module (assuming it logs with a prefix of 'teryt').   
       } else {    
          setLogInfo( "External Web Service User not Configured, Using Local Data Mode.");  
      };       
         logger("app", `--- END OF CONFIGURATION REPORT ---`);  // Logger function in app module. Assuming it logs with a prefix of 'app'.   
}    
      
function isSQLiteInConfig(): boolean { return "sqlite" in config.DATABASE_URL; }  
// Function to check if SQLITE exists within the DATABSE URL string (assumes this function returns true or false). 
        
const _validateSecurity = () =>{    // Assuming that validation functions use log as their argument and return void/Promise<void>.    
      isInsecureDefaultsNotProd() ? setLogWarning("Using DEFAULT SECRET_KEY, Change this before moving to production.") : null;  
       if(!config.CRM_*EncryptionKey()) {  // Assuming CRM* properties are not defined or empty (assumes they return false).   
           isDefaultAdminPassword() ? setLogWarning("Using DEFAULT ADMIN PASSWORD, Change this before moving to production.") : null;  
       } else{    
          logger('security', 'Security: Production mode confirmed. Mandatory keys validated by config loader');  // Logger function in security module (assuming it logs with a prefix of "security").   
      };       
}        
const isInsecureDefaultsNotProd = () => !config.IS_PROD;  
// Function to check if IS PRODUCTION not set or equal 'dev' and return true/false depending on the condition (assumes this function returns boolean). 
       const _validateTerytConfig =  (_log: logger) =>  { /* Assuming that TERYT related functions use log as their argument. */ }

File: app/config_validation.py
import logging
from app import config

logger = logging.getLogger("app.config_validation")

def report_startup_config():
    """Generates a summary of critical configuration settings at startup."""
    
    log = logging.getLogger("startup_config")
    
    log.info("--- STARTUP CONFIGURATION REPORT ---")
    log.info(f"Environment: {config.ENV.upper()}")
    log.info(f"App Display Name: {config.APP_DISPLAY_NAME}")
    
    # Database
    db_type = "SQLite" if "sqlite" in config.DATABASE_URL else "Other"
    log.info(f"Database: {db_type}")
    
    # Security Checks
    _validate_security(log)
    
    # TERYT
    _validate_teryt_config(log)
    
    log.info("--- END OF CONFIGURATION REPORT ---")

def _validate_security(log):
    # Check for insecure defaults in non-prod
    if not config.IS_PROD:
        if config.SECRET_KEY == "dev-secret-key-replace-in-prod":
            log.warning("Security: Using DEFAULT SECRET_KEY. Change this before moving to production.")
        if config.CRM_ENCRYPTION_KEY == "dev-encryption-key-must-be-32-base64-bytes==":
            log.warning("Security: Using DEFAULT ENCRYPTION_KEY. Change this before moving to production.")
        if config.CRM_ADMIN_PASSWORD == "test-change-me":
            log.warning("Security: Using DEFAULT ADMIN PASSWORD. Change this before moving to production.")
    else:
        log.info("Security: Production mode confirmed. Mandatory keys validated by config loader.")

def _validate_teryt_config(log):
    if not config.TERYT_WS_USER:
        log.info("TERYT: External WebService user not configured. Using local data mode.")
    else:
        log.info(f"TERYT: WebService user configured: {config.TERYT_WS_USER}")
